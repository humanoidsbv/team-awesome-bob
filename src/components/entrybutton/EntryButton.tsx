import React from "react";

import * as Styled from "./EntryButton.styled";

interface EntryButtonProps {
  handleClick: Function;
  isVisible?: boolean;
  label: string;
  type?: "AddButton" | "CancelButton";
  width?: number;
}
function EntryButton({
  handleClick,
  isVisible = true,
  label,
  type = "AddButton",
  width = 300,
}: EntryButtonProps) {
  return (
    <>
      {type === "AddButton" ? (
        <Styled.AddButton onClick={handleClick} isVisible={isVisible} width={width}>
          {label}
        </Styled.AddButton>
      ) : (
        <Styled.CancelButton onClick={handleClick} isVisible={isVisible} width={width}>
          {label}
        </Styled.CancelButton>
      )}
    </>
  );
}

export default EntryButton;
