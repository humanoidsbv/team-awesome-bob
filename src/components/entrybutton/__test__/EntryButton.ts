// /* eslint-disable import/no-extraneous-dependencies */
// import React from "react";
// import renderer from "react-test-renderer";
// import EntryButton from "../EntryButton";

// it("renders correctly", () => {
//   const tree = renderer.create(<EntryButton label="Test" />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

it("equals 10", () => {
  const multiplication = 2 * 5;
  expect(multiplication).toBe(10);
});
