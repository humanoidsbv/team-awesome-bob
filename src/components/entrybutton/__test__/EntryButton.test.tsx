/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import renderer from "react-test-renderer";

import EntryButton from "../EntryButton";

// it("should result in the entryButton being invisible", () => {
//   EntryButton
// });

it("renders correctly", () => {
  const entryButtonTree = renderer
    .create(<EntryButton handleClick={() => {}} isVisible type="AddButton" label="test" />)
    .toJSON();
  expect(entryButtonTree).toMatchSnapshot();
});

// function mockHandleClick() {
//   return true;
// }

// it("checks initial isVisible state", () => {

//   expect(isVisible).toBe(true);

// });

// it("checks if click is handled", () => {});
