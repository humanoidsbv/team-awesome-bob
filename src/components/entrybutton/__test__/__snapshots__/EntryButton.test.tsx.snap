// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="EntryButtonstyled__AddButton-sc-1gp19ln-0 kIPdKW"
  onClick={[Function]}
  width={300}
>
  test
</button>
`;
